### ISSUE TITLE
Does not support chinese?

### ISSUE THREAD

**Issue by greister** [hecrj/iced#213]
I test the example/todus, found this example doesn't support chinese?
Or my config file wrong ?

**Comment by hecrj** [comment]
We do not support any form of complex text layout (i.e. shaping) yet. See #33 and #208.

**Comment by cossonleo** [comment]
One way, You can pass Appication run  setting param with  'default_font' setted self font file bytes, like this: 
![图片](https://user-images.githubusercontent.com/20379044/75939476-46f12b00-5ec5-11ea-96cc-db19f42bd7df.png)

**Comment by usagi** [comment]
I have a similar issue too. And, @cossonleo 's method is not work for my testing.

The input text: `てつさび(ja) 鐵鏽(tw) 铁锈(cn) 녹(ko) Σκουριά(el) ржавчина(ru) Médecine(fr) ยา(th)`

The result:

![image](https://user-images.githubusercontent.com/115972/76273869-55bf4f80-62c2-11ea-915b-b03c04ec374a.png)

Modified code (`example/todos/main.rs`):

```rs
pub fn main() {
  Todos::run(Settings {
    default_font: Some(include_bytes!(
      "C:/Users/usagi/tmp/NotoSansCJKjp-hinted/NotoSansCJKjp-Medium.otf"
    )),
    ..Settings::default()
  })
}
```

**Comment by hecrj** [comment]
@usagi We do not support font fallback yet either, but it's in the roadmap! See the issues I previously mentioned: #33 and #208.

**Comment by usagi** [comment]
@hecrj Ah, "but it's in the roadmap!" that's good! Thank you. 👍👍

**Comment by Folyd** [comment]
@usagi I don't know why the format `ttf` is works but others aren't. You can use `.ttf` instead `.otf` or `.ttc`.

**Comment by usagi** [comment]
@Folyd Hmm, you are right. I tried three file types.

- ".otf" NG 🙅‍♀️
- ".ttc" NG 🙅‍♀️
- ".ttf" OK 🙆‍♀️

![image](https://user-images.githubusercontent.com/115972/76822909-d13a7700-6855-11ea-8396-7602b8bfbf14.png)

I don't know the details of font file types. However, it may be helpful that only the TTF type to fix the problem quickly in now.

**Comment by hecrj** [comment]
Yes, we only support TrueType fonts for now. This is a known limitation of the renderer, which uses [`rusttype`](https://gitlab.redox-os.org/redox-os/rusttype) indirectly.

**Comment by aisq2008** [comment]
let setting = Settings {
        window: window::Settings  {
            size:(800,600),
            resizable: true,
            decorations: true,
        },
        flags:(),
        default_font:Some(include_bytes!("../fonts/fang_song.ttf")),  //设置中文的.ttf字体可以支持中文
        antialiasing:true
    };

    Counter::run(setting);

   可以从C:\Windows\Fonts 中找宋体的ttf字体设置

**Comment by balthild** [comment]
@aisq2008 

Be caution. The fonts shipped with Windows is copyrighted. Bundling them in your program leads to legal issues.

You can use free (free as freedom, not free beer) fonts, such as WenQuanYi fonts instead, or load them at runtime.

**Comment by aisq2008** [comment]
@balthild i got it,thank you!

**Comment by ccsnake** [comment]
> @Folyd Hmm, you are right. I tried three file types.
> 
> * ".otf" NG 🙅‍♀️
> * ".ttc" NG 🙅‍♀️
> * ".ttf" OK 🙆‍♀️
> 
> ![image](https://user-images.githubusercontent.com/115972/76822909-d13a7700-6855-11ea-8396-7602b8bfbf14.png)
> 
> I don't know the details of font file types. However, it may be helpful that only the TTF type to fix the problem quickly in now.

hi, what's you code font? i like it.

**Comment by AAlieZZ** [comment]
> ```
> let setting = Settings {
>     window: window::Settings  {
>         size:(800,600),
>         resizable: true,
>         decorations: true,
>     },
>     flags:(),
>     default_font:Some(include_bytes!("../fonts/fang_song.ttf")),  //设置中文的.ttf字体可以支持中文
>     antialiasing:true
> };
> 
> Counter::run(setting);
> ```
> 
> 可以从C:\Windows\Fonts 中找宋体的ttf字体设置

此方法在新版本中似乎不适用了
```
error[E0308]: mismatched types
  --> src/main.rs:14:22
   |
14 | ...fault_font:Some(include_bytes!("../fonts/PingFang-Bold.ttf")),  //设置中文的.ttf字   ...
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Font`, found `Option<&...>`
   |
   = note: expected struct `Font`
                found enum `Option<&[u8; 10923212]>`
```

**Comment by kindabit** [comment]
@AAlieZZ 现在的版本可以这样加载字体：
```
  iced::application("xxx", RootWidget::update, RootWidget::view)
    .font(include_bytes!("./assets/fzss.ttf"))
    .default_font(Font::with_name("方正书宋_GBK"))
    .run_with(|| (RootWidget::new(config, i18n), Task::none()))?;
```
